### PR TITLE
feature: set query queue

### PR DESCRIPTION
**Pull request by dspo**
#### What type of this PR

Add one of the following kinds:
/kind feature

#### What this PR does / why we need it:
make an queuesto control concurrency

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |              |
| 🇨🇳 中文    |              |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sfwn — APPROVED** [review]

File: modules/cmp/metrics/metrics.go [hunk 1]
```diff
@@ -85,12 +97,12 @@ type Interface interface {
...
 func (m *Metric) query(ctx context.Context, key string, req *pb.QueryWithInfluxFormatRequest) (*pb.QueryWithInfluxFormatResponse, error) {
+	queryQueue <- struct{}{}
 	v, err := m.Metricq.QueryWithInfluxFormat(ctx, req)
-
 	if err != nil {
 		return nil, err
 	}
-
+	<-queryQueue
 	values, err := cache.MarshalValue(v)
```
Comment (dspo) [inline modules/cmp/metrics/metrics.go:106]: handle error

File: modules/cmp/metrics/metrics.go [hunk 2]
```diff
@@ -85,12 +97,12 @@ type Interface interface {
...
 	cache.FreeCache.Set(key, values, time.Now().UnixNano()+int64(time.Second*30))
```
Comment (dspo) [inline modules/cmp/metrics/metrics.go:107]: handle error

Comment (dspo) [inline modules/cmp/metrics/metrics.go:107]: `cache.FreeCache.Set(key, values, time.Now().UnixNano()+30)`

**Pull request merged**